File: app/src/main/java/ru/coffeeplanter/androidlogin/presentation/activities/MainActivityView.java
package ru.coffeeplanter.androidlogin.presentation.activities;

import ru.coffeeplanter.androidlogin.presentation.BaseView;

/**
 * Created by Ilya Solovyov on 18.09.2017.
 * is3k@ya.ru
 */

@SuppressWarnings("SpellCheckingInspection")
interface MainActivityView extends BaseView {

    void addLoginFragment();
    void addLoggedInFragment(String login);
    void replaceFromLoginToLoggedInFragment(String login);
    void replaceFromLoggedInToLoginFragment();
    void registerTimeOutReceiver();
    void unregisterTimeOutReceiver();
    void chooseFragmentOnBroadcastReceived();
    void processBackPressed();

}

File: app/src/main/java/ru/coffeeplanter/androidlogin/presentation/activities/MainActivityView.kt
package ru.coffeeplanter.androidlogin.presentation.activities

import ru.coffeeplanter.androidlogin.presentation.BaseView

/**
 * Created by Ilya Solovyov on 18.09.2017.
 * is3k@ya.ru
 */

internal interface MainActivityView : BaseView {

    fun addLoginFragment()
    fun addLoggedInFragment(login: String)
    fun replaceFromLoginToLoggedInFragment(login: String)
    fun replaceFromLoggedInToLoginFragment()
    fun registerTimeOutReceiver()
    fun unregisterTimeOutReceiver()
    fun chooseFragmentOnBroadcastReceived()
    fun processBackPressed()

}
